molo/core/views.py

```diff
@@ -10,3 +12,44 @@ def locale_set(request, locale):
 
 def health(request):
     return HttpResponse(status=200)
+
+
+def versions(request):
+    comparison_url = "https://github.com/praekelt/%s/compare/%s...%s"
+    packages = [('molo.core', 'Molo'), ('molo.profiles', 'Profiles'),
+                ('molo.commenting', 'Commenting'), ('molo.polls', 'Polls'),
+                ('molo.yourwords', 'YourWords'), ('molo.servicedirectory',
+                'Srvice Directory')]
+    packages_info = []
+    for package in packages:
+        try:
+            package_version = (
+                pkg_resources.get_distribution(package[0])).version
+
+            pypi_version = get_pypi_version(package[0])
+
+            if package[0] == 'molo.core':
+                compare_versions_link = comparison_url % (
+                    'molo', package_version, pypi_version)
+            else:
+                compare_versions_link = comparison_url % (
+                    package[0], package_version, pypi_version)
+
+            packages_info.append((package[1], pypi_version,
+                                  package_version, compare_versions_link))
+        except:
+            pypi_version = get_pypi_version(package[0])
+            packages_info.append((package[1], pypi_version, "-",
+                                 ""))
+
+    return render(request, 'admin/versions.html', {
+        'packages_info': packages_info,
+    })
+
+
+def get_pypi_version(package_name):
+    url = "https://pypi.python.org/pypi/%s/json"
+    content = (requests.get(url % package_name)).content
+    content = content[((content.find('version')) + 11):]
```

Comment:
There should be a cleaner way of using this. I'll dig around and give you a suggestion